api/config/v1alpha1/envoygateway_types.go

```diff
@@ -313,6 +320,35 @@ type ExtensionTLS struct {
 	CertificateRef gwapiv1b1.SecretObjectReference `json:"certificateRef"`
 }
 
+// EnvoyGatewayObservability defines the Envoy Gateway Observability configuration.
+type EnvoyGatewayObservability struct {
+
+	// Debug defines Debug Server configuration of Envoy Gateway
+	//
+	// +optional
+	Debug *DebugConfig `json:"debug,omitempty"`
+}
+
+// DebugConfig defines the Envoy Gateway Debug Server configuration.
+type DebugConfig struct {
+	// IPNetworkFamily defines the debug server IPNetworkFamily.
+	//
+	// +optional
+	// +kubebuilder:default=tcp
+	IPNetworkFamily string `json:"ipNetworkFamily,omitempty"`
```

Comment:
imo we should only expose `Port` for now, and internally bind to `tcp:127.0.0.1:<port>`